### DISCUSSION TITLE
Change in Browser flow for user migration

### DISCUSSION THREAD

**Discussion by Bhavik237** [keycloak/keycloak#21093] [Q&A]
Greetings, everyone!
I am currently performing user data migration without any set credentials. Instead, I will assign the attribute "toBeReset = yes" to each user and follow the flow depicted in the following images:

User migration flow:
![User migration (2)](https://github.com/keycloak/keycloak/assets/57186072/e394f92d-76b8-4eaf-8735-b91c90404c97)

![Screenshot from 2023-06-20 09-36-07](https://github.com/keycloak/keycloak/assets/57186072/6a10a01c-b3d4-42d4-99f3-42e092b43ade)
Browser flow:
![Screenshot from 2023-06-20 09-36-25](https://github.com/keycloak/keycloak/assets/57186072/7109b82b-da49-4816-b892-e154bf1ac5c3)

**However, I am facing an issue where, after sending the reset password email to the user, it displays the username form instead of proceeding to the password form as mentioned in the flow.**

Thanks

**Comment by darius-m** [comment]
If you simply need to ask the user to change their password on the first login, you should be able to use the "Update password" required action. After the user logs in for the first time, they will be required to change their password before being able to continue.

**Reply by Bhavik237** [reply]
A simple password update will not be effective because if the user enters their username, they will be directed straight to the set password screen since their credentials are not set.
